CRM/Contact/Page/AJAX.php

```diff
@@ -467,30 +467,21 @@ public static function getContactPhone() {
     $phoneTypes = CRM_Core_OptionGroup::values('phone_type', TRUE, FALSE, FALSE, NULL, 'name');
     $mobileType = CRM_Utils_Array::value('Mobile', $phoneTypes);
 
-    $name = CRM_Utils_Array::value('name', $_GET);
+    $name = CRM_Utils_Request::retrieveValue('name', 'String', NULL, FALSE, 'GET');
     if ($name) {
-      $name = CRM_Utils_Type::escape($name, 'String');
       $queryString = " ( cc.sort_name LIKE '%$name%' OR cp.phone LIKE '%$name%' ) ";
```

Comment:
I really don't think we should be concatenating string values from the url into a query. I think this line ought to be using `%1` `%2` etc placeholders and we should be feeding values into an array as the 2nd param to `executeQuery`.